chatlas/_auto.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import Optional
+
+from ._anthropic import ChatAnthropic, ChatBedrockAnthropic
+from ._chat import Chat
+from ._github import ChatGithub
+from ._google import ChatGoogle
+from ._groq import ChatGroq
+from ._ollama import ChatOllama
+from ._openai import ChatAzureOpenAI, ChatOpenAI
+from ._perplexity import ChatPerplexity
+from ._turn import Turn
+
+_provider_chat_model_map = {
+    "anthropic": ChatAnthropic,
+    "bedrock:anthropic": ChatBedrockAnthropic,
+    "github": ChatGithub,
+    "google": ChatGoogle,
+    "groq": ChatGroq,
+    "ollama": ChatOllama,
+    "azure:openai": ChatAzureOpenAI,
+    "openai": ChatOpenAI,
+    "perplexity": ChatPerplexity,
+}
+
+
+def ChatAuto(
+    provider: Optional[str] = None,
+    system_prompt: Optional[str] = None,
+    turns: Optional[list[Turn]] = None,
+    **kwargs,
+) -> Chat:
+    """
+    Factory function to create a Chat instance based on a provider specified in code or in an environment variable.
+
+    This function creates a Chat instance based on the specified provider, with optional system prompt and conversation turns.
+    The provider can be specified either through the function parameter or via the CHATLAS_CHAT_PROVIDER environment variable.
+    Additional configuration can be provided through kwargs or the CHATLAS_CHAT_ARGS environment variable (as JSON). This allows
+    you to easily switch between different chat providers by changing the environment variable without modifying your code.
+
```

Comment:
I think here would be a good place to explain environment variables will win over `kwargs`